sound/soc/sof/ipc4-loader.c

```diff
@@ -186,6 +187,13 @@ static void sof_ipc4_query_fw_configuration(struct snd_sof_dev *sdev)
 		case SOF_IPC4_FW_CFG_TRACE_LOG_BYTES:
 			dev_vdbg(sdev->dev, "Trace log size: %u\n", *tuple->value);
 			break;
+		case SOF_IPC4_FW_CFG_MAX_LIBS_COUNT:
+			ipc4_data->max_fw_libs = *tuple->value;
+			ipc4_data->fw_lib_names = devm_kcalloc(sdev->dev, ipc4_data->max_fw_libs,
+							       LIBRARY_FILENAME_LEN, GFP_KERNEL);
+
+			/* TODO: check if mem alloc is successful and return err */
```

Comment:
Should we fixup! the fw_loader_ops and make the `query_fw_configuration()` to return `int` to be able to fail?